### PR TITLE
Fixes and tweaks

### PR DESCRIPTION
**Pull request by harshad1**
Several fixes and tweaks

1. Reduced horizontal scroll sensitivity in MainActivity
2. Fixed share-into formatting and insertion
3. Keyboard state preserved on template insertion (like several other dialogs)
4. Fixing folder opening from widgets
5. Several other small tweaks and fixes

### CODE REVIEW

File: app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java [hunk 1]
```diff
@@ -213,6 +216,23 @@ private void appendToExistingDocument(final File file, final String separator, f
...
+            // Interpolate parts
+            final long time = System.currentTimeMillis();
+            for (int i = 0 ; i < parts.size(); i++) {
+                parts.set(i, _cu.formatDateTime(context, parts.get(i), time));
+            }
+
+            // Put the shared text in the right place
+            parts.add(1, shared);
+
+            return String.join("", parts);
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java:233]: Fix share into interpolation / formatting

File: app/src/main/java/net/gsantner/markor/activity/openeditor/OpenFromShortcutOrWidgetActivity.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        final Intent newIntent = new Intent(intent);
+        final File intentFile = (File) intent.getSerializableExtra(Document.EXTRA_PATH);
+        if (intentFile.isDirectory()) {
+            newIntent.setClass(this, MainActivity.class);
+            startActivity(newIntent);
+        } else {
+            newIntent.setClass(this, DocumentActivity.class);
+            DocumentActivity.launch(this, null, null, newIntent, null);
+        }
+        finish();
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/activity/openeditor/OpenFromShortcutOrWidgetActivity.java:42]: Send intent to MainActivity or DocumentActivity as appropriate - without this, launching folders from the widget was broken

File: app/src/main/java/net/gsantner/markor/format/asciidoc/AsciidocSyntaxHighlighter.java [hunk 3]
```diff
@@ -108,39 +108,39 @@ public class AsciidocSyntaxHighlighter extends SyntaxHighlighterBase {
...
 
 
-image:https://personal.sron.nl/~pault/images/scheme_dark.png[Dark scheme]
+    image:https://personal.sron.nl/~pault/images/scheme_dark.png[Dark scheme]
 
-Colours: '#222255', '#225555', '#225522', '#666633', '#663333', '#555555'.
+    Colours: '#222255', '#225555', '#225522', '#666633', '#663333', '#555555'.
 
-*/
+    */
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/format/asciidoc/AsciidocSyntaxHighlighter.java:143]: Just formatting.

File: app/src/main/java/net/gsantner/markor/model/AppSettings.java [hunk 4]
```diff
@@ -167,7 +167,7 @@ public boolean isHighlightCodeMonospaceFont() {
     }
 
     public boolean isHighlightCodeBlock() {
-        return !getBool(R.string.pref_key__hightlight_code_block_disabled, false);
+        return !getBool(R.string.pref_key__highlight_code_block_disabled, false);
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/model/AppSettings.java:170]: Typo in key name

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -454,7 +454,7 @@ work. If not, see <https://creativecommons.org/publicdomain/zero/1.0/>.
     <string name="match_all">Match All</string>
     <string name="dynamic_notebook_root_links">Dynamic search for notebook root</string>
     <string name="search_dynamically_for_notebook_root">In order to follow wiki links, dynamically search for the notebook.zim file to identify the root directory for the current notebook. Check this if you use multiple zim notebooks or the notebook directory set in the app does not correspond to your zim notebook.</string>
-	<string name="share_into_format_prefix_for_text_shared">Format/Prefix for text shared into Markor. Follows Android\'s SimpleDateFormat rules.\n\nUse the placeholder \'%s\' (surrounded by single quotes) to position the shared text. If not specified, the shared text is appended and the format used as a prefix.</string>
+	<string name="share_into_format_prefix_for_text_shared">Format/Prefix for text shared into Markor. Follows Android\'s SimpleDateFormat rules.\n\nUse the placeholder {{text}} to position the shared text. If not specified, the shared text is appended and the format used as a prefix.</string>
```
Comment (harshad1) [inline app/src/main/res/values/strings.xml:457]: We use {{text}} instead of '%s' now. Updating to match the actual implementation. This will need to updated in the other translations too

File: app/thirdparty/res/xml/markor_widget.xml [hunk 6]
```diff
@@ -16,6 +16,8 @@
     android:initialLayout="@layout/widget_layout"
     android:minWidth="100dp"
     android:minHeight="200dp"
+    android:targetCellHeight="2"
+    android:targetCellWidth="2"
```
Comment (harshad1) [inline app/thirdparty/res/xml/markor_widget.xml:20]: Default widget is small

File: app/thirdparty/res/layout/widget_file_item.xml [hunk 7]
```diff
@@ -29,12 +30,13 @@
...
         android:ellipsize="end"
         android:gravity="center|start"
         android:minHeight="19sp"
         android:singleLine="true"
         android:textColor="@color/primary_text"
-        android:textSize="16sp" />
+        android:textSize="16sp"
+        android:drawablePadding="4dp"
+        app:drawableStartCompat="@drawable/ic_file_gray_24dp"
+        app:drawableLeftCompat="@drawable/ic_file_gray_24dp" />
```
Comment (harshad1) [inline app/thirdparty/res/layout/widget_file_item.xml:41]: Fixed drawable padding and using compat like android studio recommends

**Comment by harshad1** [comment]
@gsantner I ran into a weird issue when debugging - a fresh install of mordor through android studio kept crashing on launch at the file permissions dialog. I had to go into settings and explicitly give mordor the all files permission. This doesn't seem to affect prod builds.

**Comment by min7-i** [comment]
> I sometimes hear about this issue, so also on release build. But not occured yet here. So would be good to know why permission dialog not triggers, and why it crashes

Hopefully I'm not hijacking the PR here. I just noticed your comments about the missing file permissions dialog which I experienced in the past days as well when I installed Mordor (v2.10.6-2322) on my phone (android 9). Afterwards I tested the installation on my new phone (android 13) where the file permissions dialog was correctly displayed after the installation. So maybe it is related to the android version?

If I can provide any additional information that might help, just let me know.

**Comment by harshad1** [comment]
@min7-i if you are able to reproduce and get a stacktrace that would be great. My current phone runs Android 13 so it is not restricted to earlier versions of Android

**Comment by min7-i** [comment]
@harshad1 I haven't captured a log on an Android device before, so I'm not sure if this is exactly what you're looking for, but here's the log file from the installation of mordor, the initial start of the app and the crash afterwards.
[mordor_log.txt](https://github.com/gsantner/markor/files/10239197/mordor_log.txt)

File: app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java [hunk 8]
```diff
@@ -722,15 +723,21 @@ public boolean isStateBad() {
...
             errorClipText();
             return false;
         }
 
         // Document is written iff writeable && content has changed
         final CharSequence text = _hlEditor.getText();
         if (!_document.isContentSame(text)) {
-            if (_document.saveContent(getActivity(), text, _cu, forceSaveEmpty)) {
+            final int minLength = GsContextUtils.TEXTFILE_OVERWRITE_MIN_TEXT_LENGTH;
+            if (!forceSaveEmpty && text != null && text.length() < minLength) {
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java:736]: Without this saving a small file (< 2 chars) would be treated as an error and the characters would be copied to the clipboard.

File: app/src/main/java/net/gsantner/markor/activity/MainActivity.java [hunk 9]
```diff
@@ -104,45 +105,22 @@ public void onPageSelected(int position) {
...
-        final File dir = getIntentDir(intent, null);
+        final File dir = GsFileUtils.getValidIntentDir(intent, null);
         final GsFileBrowserFragment frag = getNotebook();
         if (frag != null && dir != null) {
             frag.post(() -> frag.getAdapter().setCurrentFolder(dir));
             _bottomNav.postDelayed(() -> _bottomNav.setSelectedItemId(R.id.nav_notebook), 10);
         }
     }
 
-    private static File getIntentDir(final Intent intent, final File fallback) {
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/activity/MainActivity.java:123]: Moved to GsFileUtils

File: app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java (hunk 1 above)
Comment (gsantner) [reply app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java:233]: I recall String.join throwing exception on older android devices. There should be some TextUtils methods as alternative around

File: app/src/main/java/net/gsantner/markor/activity/MainActivity.java [hunk 10]
```diff
@@ -104,45 +109,46 @@ public void onPageSelected(int position) {
...
+            recyclerView.setScrollingTouchSlop(RecyclerView.TOUCH_SLOP_PAGING);
+            final Field touchSlopField = RecyclerView.class.getDeclaredField("mTouchSlop");
+            touchSlopField.setAccessible(true);
+            final int touchSlop = (int) touchSlopField.get(recyclerView);
+            touchSlopField.set(recyclerView, touchSlop * SLOP_MULTIPLIER);
+        } catch(NoSuchFieldException e) {
+            Log.d(MainActivity.class.getName(), e.getMessage());
+        } catch(IllegalAccessException e) {
+            Log.d(MainActivity.class.getName(), e.getMessage());
+        } catch (NullPointerException e) {
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/activity/MainActivity.java:136]: Enough to have one catchall. Not wanted to crash something here for that hack. And whatever error happens, it means the workaround doesn't work anymore and all we can do is print a warning

File: app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java [hunk 11]
```diff
@@ -752,4 +754,32 @@ public static List<File> replaceFilesWithCachedVariants(@Nullable List<File> fil
         }
         return files;
     }
+
+    // Get intent file
+    public static File getIntentFile(final Intent intent, final File fallback) {
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java:759]: please have that one in ./app/src/main/java/net/gsantner/markor/util/MarkorContextUtils.java

File: app/src/main/java/net/gsantner/markor/activity/MainActivity.java (hunk 9 above)
Comment (gsantner) [reply app/src/main/java/net/gsantner/markor/activity/MainActivity.java:123]: please have that one in ./app/src/main/java/net/gsantner/markor/util/MarkorContextUtils.java

**Comment by harshad1** [comment]
@gsantner Addressed comments.

File: app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java [hunk 12]
```diff
@@ -174,7 +175,7 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         webSettings.setAllowFileAccess(true);
         webSettings.setAllowContentAccess(true);
         webSettings.setAllowFileAccessFromFileURLs(true);
-        webSettings.setAllowUniversalAccessFromFileURLs(true);
+        webSettings.setAllowUniversalAccessFromFileURLs(false);
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java:178]: This can break somebody's way to use Markor. Why the change?

**Review by gsantner — APPROVED** [review]

**Comment by gsantner** [comment]
Thanks! Merging!
